test/fw/bin/pbs_sys_report

```diff
@@ -42,22 +42,56 @@ import csv
 import json
 import sys
 import os
+import getopt
 
 
-def write_to_csv(head, csvrows, i):
-    with open('sys_report.csv', 'a+') as fp:
+def usage():
+    msg = []
+    msg += ['Usage: ' + os.path.basename(sys.argv[0])]
+    msg += [' [PTL_JSON_FILE] \n\n']
+    msg += [' System monitoring results to csv conversion tool']
+    msg += ['--output-file : path to generate file\n']
+    msg += ['--help or -h : To display usage information\n']
+    print(''.join(msg))
+
+
+def write_to_csv(head, csvrows, i, filepath):
+    with open(filepath, 'a+') as fp:
         if i == 0:
             csv.writer(fp).writerow(head)
         csv.writer(fp).writerow(csvrows)
 
 
 if __name__ == '__main__':
 
+    if len(sys.argv) < 2:
+        usage()
+        sys.exit(1)
+
+    try:
+        opts, args = getopt.getopt(sys.argv[2:], "h",
+                                   ["help", "output-file="])
```

Comment:
I still suggest looking into argparse over getopt().  It allows it wraps up the usage and all the arguments when you define the parser.  There's a thing called a positional argument you can use for the json file.  You won't have to check if the output file is specified, because you can set a default when you define the parser.

This is just a suggestion though.  I'll sign off.

Reply:
Updated file with argparse